utils/context.go

```diff
@@ -0,0 +1,31 @@
+package utils
+
+import (
+	"context"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
+)
+
+const MetadataKey = "viam-metadata"
+
+func ContextWithMetadata(ctx context.Context) (context.Context, map[string][]string) {
+	md := make(map[string][]string)
+	ctx = context.WithValue(ctx, MetadataKey, md)
```

Comment:
can we add a check that the MetadataKey is not already present? If it is, I think we can safely panic

Reply:
I ended up changing this to check if the MetadataKey is already present, and if so, return that. I also ended up using a custom type for MetadataKey so we wouldn't have conflicts with any keys set from third-party libraries/modules (it was the linter's idea, I can't take credit).